src/prefect/orion/orchestration/core_policy.py

```diff
@@ -305,6 +292,55 @@ async def before_transition(
         await self.reject_transition(state=retry_state, reason="Retrying")
 
 
+class RestartFlowRun(BaseOrchestrationRule):
+    FROM_STATES = TERMINAL_STATES
+    TO_STATES = [states.StateType.SCHEDULED]
+
+    async def before_transition(
+        self,
+        initial_state: Optional[states.State],
+        proposed_state: Optional[states.State],
+        context: FlowOrchestrationContext,
+    ):
+        if context.run.parent_task_run_id:
+            await self.abort_transition("Cannot restart a subflow run.")
```

Comment:
You should be allowed to do this e.g. when `run_deployment` has been used you could want to restart a run that has been linked. I think your next check that a deployment id is present should be sufficient here.